src/services/legal-services.ts

```diff
@@ -1,76 +1,87 @@
 // Libraries
 import { AxiosInstance as axios } from '@/utils'
 import { StatusCodes } from 'http-status-codes'
-import { DissolutionFilingIF, IncorporationFilingIF, NameRequestIF, RegistrationFilingIF } from '@/interfaces'
+import { DissolutionFilingIF, IncorporationFilingIF, NameRequestIF, RegistrationFilingIF, RestorationFilingIF }
+  from '@/interfaces'
 import { FilingTypes } from '@/enums'
 
 /**
  * Class that provides integration with the Legal API.
  */
 export default class LegalServices {
   /**
-   * Fetches a draft filing using the temp reg number.
-   * @param id the temp reg number
-   * @returns a promise to return the formatted draft filing, else exception
+   * Fetches the first or only filing.
+   * This is expected to be a draft IA or Registration.
+   * @param tempId the temp registration number
+   * @returns a promise to return the draft filing, else exception
    */
-  static async fetchDraftApplication (id: string): Promise<IncorporationFilingIF | RegistrationFilingIF> {
-    const url = `businesses/${id}/filings`
+  static async fetchFirstOrOnlyFiling (tempId: string): Promise<IncorporationFilingIF | RegistrationFilingIF> {
+    const url = `businesses/${tempId}/filings`
     return axios.get(url)
       .then(response => {
         let filing
         // if response is a list then look at only the first filing
         if (Array.isArray(response?.data?.filings)) filing = response.data.filings[0].filing
         // otherwise expect response to be a single filing
         if (response?.data?.filing) filing = response.data.filing
-        const filingName = filing?.header?.name
+
+        const filingName = filing?.header?.name as FilingTypes
         const filingId = +filing?.header?.filingId || 0
 
         if (!filing || !filingName || !filingId) {
           throw new Error('Invalid API response')
         }
 
-        switch (filingName) {
-          case FilingTypes.INCORPORATION_APPLICATION:
-            return this.formatEmptyIncorporationApplication(filing)
-          case FilingTypes.REGISTRATION:
-            return this.formatEmptyRegistration(filing)
-          default:
-            throw new Error('Invalid filing name')
-        }
+        return filing
```

Comment:
Renamed and generalized to fetch draft IA or Registration from `filings` endpoint.

The previous filing-type-specific code was moved to App.vue.